server/liveblog/blogs/tasks.py

```diff
@@ -0,0 +1,84 @@
+import io
+import logging
+
+import superdesk
+from celery.exceptions import SoftTimeLimitExceeded
+from flask import current_app as app
+from settings import S3_CELERY_COUNTDOWN, S3_CELERY_MAX_RETRIES
+from superdesk import get_resource_service
+from superdesk.celery_app import celery
+from superdesk.notification import push_notification
+
+from .embeds import embed
+from .exceptions import MediaStorageUnsupportedForBlogPublishing
+from .utils import check_media_storage, get_blog_path
+
+logger = logging.getLogger('superdesk')
+
+
+def publish_embed(blog_id, api_host=None, theme=None):
+    # Get html using embed() blueprint
+    html = embed(blog_id, api_host, theme)
+    check_media_storage()
+    file_path = get_blog_path(blog_id)
+    # Remove existing file
+    app.media.delete(app.media.media_id(file_path, version=False))
+    logger.warning('Embed file "{}" for blog "{}" removed from s3'.format(file_path, blog_id))
+    # Upload new file
+    file_id = app.media.put(io.BytesIO(bytes(html, 'utf-8')), filename=file_path, content_type='text/html',
+                            version=False)
+    logger.warning('Embed file "{}" for blog "{}" uploaded to s3'.format(file_path, blog_id))
+    return superdesk.upload.url_for_media(file_id)
+
+
+def delete_embed(blog_id):
+    check_media_storage()
+    file_path = get_blog_path(blog_id)
+    # Remove existing file
+    app.media.delete(file_path)
+
+
+def _publish_blog_embed_on_s3(blog_id, safe=True):
```

Comment:
Don't really understand why you would be put this in a separate function unless you intend to reuse it